File: challenge-310/pokgopun/python/ch-1.py
### https://theweeklychallenge.org/blog/perl-weekly-challenge-310/
"""

Task 1: Arrays Intersection

Submitted by: [38]Mohammad Sajid Anwar
     __________________________________________________________________

   You are given a list of array of integers.

   Write a script to return the common elements in all the arrays.

Example 1

Input: $list = ( [1, 2, 3, 4], [4, 5, 6, 1], [4, 2, 1, 3] )
Output: (1, 4)

Example 2

Input: $list = ( [1, 0, 2, 3], [2, 4, 5] )
Output: (2)

Example 3

Input: $list = ( [1, 2, 3], [4, 5], [6] )
Output: ()

Task 2: Sort Odd Even
"""
### solution by pokgopun@gmail.com

from itertools import chain

def ai(tup: tuple[tuple[int]]) -> tuple[int]:
    l = len(tup)
    if l == 0:
        return tuple()
    t = set(tup[0])
    for tr in tup[1:]:
        t.intersection_update(set(tr))
    return tuple(t)

import unittest

class TestAI(unittest.TestCase):
    def test(self):
        for inpt, otpt in {
                ( (1, 2, 3, 4), (4, 5, 6, 1), (4, 2, 1, 3) ): (1, 4),
                ( (1, 0, 2, 3), (2, 4, 5) ): (2,),
                ( (1, 2, 3), (4, 5), (6,) ): (),
                }.items():
            self.assertEqual(ai(inpt),otpt)

unittest.main()

File: challenge-330/pokgopun/perl/ch-1.pl
### https://theweeklychallenge.org/blog/perl-weekly-challenge-330/
=begin

Task 1: Clear Digits

Submitted by: [44]Mohammad Sajid Anwar
     __________________________________________________________________

   You are given a string containing only lower case English letters and
   digits.

   Write a script to remove all digits by removing the first digit and the
   closest non-digit character to its left.

Example 1

Input: $str = "cab12"
Output: "c"

Round 1: remove "1" then "b" => "ca2"
Round 2: remove "2" then "a" => "c"

Example 2

Input: $str = "xy99"
Output: ""

Round 1: remove "9" then "y" => "x9"
Round 2: remove "9" then "x" => ""

Example 3

Input: $str = "pa1erl"
Output: "perl"

Task 2: Title Capital
=end
### solution by pokgopun@gmail.com
